doc/api/assert.md

```diff
@@ -2548,6 +2548,237 @@ assert.throws(throwingFirst, /Second$/);
 Due to the confusing error-prone notation, avoid a string as the second
 argument.
 
+## `assert.deepMatch(actual, expected[, message])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `actual` {any}
+* `expected` {any}
+* `message` {string|Error}
+
+[`assert.deepMatch()`][] evaluates the equivalence between the `actual` and `expected` parameters by
+performing a deep comparison. This function ensures that all properties defined
+in the `expected` parameter match those in the `actual` parameter in
+both value and type, allowing type coercion.
+
+```mjs
+import assert from 'node:assert';
+
+assert.deepMatch({ a: 1, b: '2' }, { a: 1, b: 2 });
+// OK
```

Comment:
How is this different from `deepStrictEqual(...)` and `deepEqual(...)`? The docs should expand on that more. As it is, these examples do not really provide enough detail to determine the correct one to choose.

Reply:
thanks for the update! I adjusted the docs now